packages/ui/src/components/SelectInputV2/Dropdown.tsx

```diff
@@ -0,0 +1,916 @@
+  selectedValues: OptionType[]
+  setIsDropdownVisible: Dispatch<SetStateAction<boolean>>
+  onChange?: (value: string[]) => void
+  setSelectedValues: Dispatch<SetStateAction<OptionType[]>>
+  loadMore?: ReactNode
+  optionalInfoPlacement: 'left' | 'right'
+  defaultSearchValue: string | null
+  isLoading?: boolean
+  selectAll?: { label: ReactNode; description?: string }
+  allSelected: boolean
+  setAllSelected: Dispatch<SetStateAction<boolean>>
+  selectAllGroup?: boolean
+  selectedGroups: string[]
+  setSelectedGroups: Dispatch<SetStateAction<string[]>>
+  numberOfOptions: number
+}
+const CreateDropdown = ({
+  displayedOptions,
+  isEmpty,
+  emptyState,
+  descriptionDirection,
+  selectedValues,
+  setIsDropdownVisible,
+  onChange,
+  setSelectedValues,
+  loadMore,
+  optionalInfoPlacement,
+  defaultSearchValue,
+  isLoading,
+  selectAll,
+  multiselect,
+  options,
+  allSelected,
+  setAllSelected,
+  selectAllGroup,
+  selectedGroups,
+  setSelectedGroups,
+  numberOfOptions,
+}: CreateDropdownProps) => {
+  const focusedItemRef = useRef<HTMLButtonElement>(null)
+
+  useEffect(() => {
+    if (defaultSearchValue && focusedItemRef?.current) {
+      focusedItemRef.current.focus()
+    }
+  }, [defaultSearchValue])
+
+  if (isEmpty) {
+    return (
+      <EmptyState gap={2} alignItems="center">
+        {emptyState ?? (
+          <Text variant="bodyStrong" as="p">
+            No options
+          </Text>
+        )}
+      </EmptyState>
+    )
+  }
+
+  const handleClick = (clickedOption: OptionType, group?: string) => {
+    if (multiselect) {
+      if (selectedValues.includes(clickedOption)) {
+        setSelectedValues(selectedValues.filter(val => val !== clickedOption))
+        onChange?.(
+          selectedValues
+            .filter(val => val !== clickedOption)
+            .map(val => val?.value),
+        )
+        setAllSelected(false)
+
+        if (selectAllGroup && group && selectedGroups.includes(group)) {
+          setSelectedGroups(
+            selectedGroups.filter(selectedGroup => selectedGroup !== group),
+          )
+        }
+      } else {
+        setSelectedValues([...selectedValues, clickedOption])
+        onChange?.([...selectedValues, clickedOption].map(val => val?.value))
+        if (selectedValues.length + 1 === numberOfOptions) {
+          setAllSelected(true)
+        }
+        if (
+          !Array.isArray(options) &&
+          group &&
+          options[group].every(
+            option =>
+              [...selectedValues, clickedOption].includes(option) ||
+              option.disabled,
+          )
+        ) {
+          setSelectedGroups([...selectedGroups, group])
+        }
+      }
+    } else {
+      setSelectedValues([clickedOption])
+      onChange?.([clickedOption.value])
+    }
+    setIsDropdownVisible(multiselect) // hide the dropdown on click when single select only
+  }
+  const displayOption = (option: OptionType) => {
```

Comment:
Is It possible to export this in a different file ? Better for maintenance and readability